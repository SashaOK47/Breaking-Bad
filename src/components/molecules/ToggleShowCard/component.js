import React from "react";
import PropTypes from "prop-types";
import style from "./styles.module.scss";

const ToggleShowCard = ({
  grid,
  clickHandler,
  clickHandlerActive,
  enterHandlerActive,
  enterHandler,
}) => {
  let classActive = grid ? style.active : "";
  let noActive = !grid ? style.active : "";
  return (
    <div className={style.icons}>
      <button type="button" onKeyDown={enterHandler}>
        <svg
          className={noActive}
          onClick={clickHandler}
          width="24"
          height="24"
          viewBox="0 0 24 24"
          fill="none"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect x="1" y="1" width="22" height="9" rx="1" strokeWidth="2" />
          <rect x="1" y="14" width="22" height="9" rx="1" strokeWidth="2" />
        </svg>
      </button>
      <button type="button" onKeyDown={enterHandlerActive}>
        <svg
          className={classActive}
          onClick={clickHandlerActive}
          width="24"
          height="24"
          viewBox="0 0 24 24"
          fill="none"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect x="1" y="1" width="9" height="9" rx="1" strokeWidth="2" />
          <rect x="1" y="14" width="9" height="9" rx="1" strokeWidth="2" />
          <rect x="14" y="1" width="9" height="9" rx="1" strokeWidth="2" />
          <rect x="14" y="14" width="9" height="9" rx="1" strokeWidth="2" />
        </svg>
      </button>
    </div>
  );
};
ToggleShowCard.propTypes = {
  grid: PropTypes.bool,
  clickHandler: PropTypes.func,
  clickHandlerActive: PropTypes.func,
  enterHandlerActive: PropTypes.func,
  enterHandler: PropTypes.func,
};
export default ToggleShowCard;
