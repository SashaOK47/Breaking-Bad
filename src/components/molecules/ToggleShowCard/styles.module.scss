@import "../../../assets/styles/main.scss";
.icons {
  max-height: 1.5rem;
  height: 100%;
  button {
    width: 1.5rem;
    height: 1.5rem;
    background: none;
    margin-left: 0.5rem;
    cursor: pointer;
    &:first-child {
      margin-left: 0;
    }
    svg {
      width: 100%;
      height: 100%;
      stroke: $gray;
      &.active {
        stroke: $colorBgLink;
      }
    }
    &:focus-visible {
      @include boxShadowFocus;
    }
  }
}
