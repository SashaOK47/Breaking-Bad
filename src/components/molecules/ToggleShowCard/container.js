import ToggleShowCard from "./component";
import { useSelector, useDispatch } from "react-redux";
import { toggleGridtoList } from "../../../store/actions/characters";

const ToggleShowCardContainer = () => {
  const dispatch = useDispatch();
  const grid = useSelector((state) => state.characters.isGrid);
  const clickHandler = () => {
    dispatch(toggleGridtoList(false));
  };
  const enterHandler = (e) => {
    if (e.key !== "Enter") return;
    dispatch(toggleGridtoList(false));
  };
  const clickHandlerActive = () => {
    dispatch(toggleGridtoList(true));
  };
  const enterHandlerActive = (e) => {
    if (e.key !== "Enter") return;
    dispatch(toggleGridtoList(true));
  };

  return (
    <ToggleShowCard
      grid={grid}
      clickHandler={clickHandler}
      clickHandlerActive={clickHandlerActive}
      enterHandler={enterHandler}
      enterHandlerActive={enterHandlerActive}
    />
  );
};
export const container = ToggleShowCardContainer;
