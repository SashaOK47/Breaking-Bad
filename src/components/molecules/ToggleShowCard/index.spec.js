import React from "react";
import ToggleShowCard from "./component";

let grid = true;
const setUp = (props) => shallow(<ToggleShowCard {...props} />);

describe("ToggleShowCard component", () => {
  it("should render ToggleShowCard component", () => {
    const component = setUp();
    expect(component).toMatchSnapshot();
  });
  it("with the transfer of prop grid", () => {
    const component = setUp({ grid });
    expect(component).toMatchSnapshot();
  });
});
