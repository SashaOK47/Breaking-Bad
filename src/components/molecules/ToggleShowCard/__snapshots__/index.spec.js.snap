// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleShowCard component should render ToggleShowCard component 1`] = `
<div
  className="icons"
>
  <button
    type="button"
  >
    <svg
      className="active"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        height="9"
        rx="1"
        strokeWidth="2"
        width="22"
        x="1"
        y="1"
      />
      <rect
        height="9"
        rx="1"
        strokeWidth="2"
        width="22"
        x="1"
        y="14"
      />
    </svg>
  </button>
  <button
    type="button"
  >
    <svg
      className=""
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        height="9"
        rx="1"
        strokeWidth="2"
        width="9"
        x="1"
        y="1"
      />
      <rect
        height="9"
        rx="1"
        strokeWidth="2"
        width="9"
        x="1"
        y="14"
      />
      <rect
        height="9"
        rx="1"
        strokeWidth="2"
        width="9"
        x="14"
        y="1"
      />
      <rect
        height="9"
        rx="1"
        strokeWidth="2"
        width="9"
        x="14"
        y="14"
      />
    </svg>
  </button>
</div>
`;

exports[`ToggleShowCard component with the transfer of prop grid 1`] = `
<div
  className="icons"
>
  <button
    type="button"
  >
    <svg
      className=""
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        height="9"
        rx="1"
        strokeWidth="2"
        width="22"
        x="1"
        y="1"
      />
      <rect
        height="9"
        rx="1"
        strokeWidth="2"
        width="22"
        x="1"
        y="14"
      />
    </svg>
  </button>
  <button
    type="button"
  >
    <svg
      className="active"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        height="9"
        rx="1"
        strokeWidth="2"
        width="9"
        x="1"
        y="1"
      />
      <rect
        height="9"
        rx="1"
        strokeWidth="2"
        width="9"
        x="1"
        y="14"
      />
      <rect
        height="9"
        rx="1"
        strokeWidth="2"
        width="9"
        x="14"
        y="1"
      />
      <rect
        height="9"
        rx="1"
        strokeWidth="2"
        width="9"
        x="14"
        y="14"
      />
    </svg>
  </button>
</div>
`;
